Fix error when using type aliases referencing other type aliases (#10624)
pydantic/pydantic: docs/errors/usage_errors.md

    assert exc_info.code == 'removed-kwargs'
```

## Circular reference schema {#circular-reference-schema}

> While Pydantic can handle recursive types, it does not allow _directly_ circular references in the schema.

Reviewer: ```suggestion
This error is raised when a circular reference is found that would otherwise result in an infinite recursion.
```

> For example, these are valid types:
> 
> ```py test="skip" lint="skip" upgrade="skip"
> type A = A | int
> 
> type B = list[C]
> type C = B | None
> ```

Reviewer: Actually this example is invalid from a type checking perspective, and it does makes sense in some way. I think the logic applied is "can we end up with a stop when trying to match a value to the type?".

For instance, with

```python
type B = list[C]
type C = B | None
```

the value `[None, [[None]], []]` successfully matches and we came to stop, thanks to the `| None` part. without it, there's no such value that can match the type alias without dealing with infinite recursion (hence my suggestion above).

Let's go with:

```suggestion
For example, this is a valid type alias:

```py test="skip" lint="skip" upgrade="skip"
type A = list[A] | None
```
```

while these are not:

```py test="skip" lint="skip" upgrade="skip"
type A = A

type B = C
type C = B
```

So the latter will raise an error:

```py
from typing_extensions import TypeAliasType

from pydantic import PydanticUserError, TypeAdapter

A = TypeAliasType('A', 'A')

B = TypeAliasType('B', 'C')
C = TypeAliasType('C', 'B')

try:
    TypeAdapter(A)
except PydanticUserError as exc_info:
    assert exc_info.code == 'circular-reference-schema'

try:
    TypeAdapter(B)
except PydanticUserError as exc_info:
    assert exc_info.code == 'circular-reference-schema'
```

> 

Reviewer: I think we can remove this part, we already described the kind of valid and invalid type aliases.
